syntax = "proto3";

package hotel;

service ReservationService {
  rpc ReserverChambre (ReservationRequest) returns (ReservationResponse);
  rpc VerifierDisponibilite (DisponibiliteRequest) returns (DisponibiliteResponse);
  rpc AnnulerReservation (AnnulationRequest) returns (AnnulationResponse);
  rpc ConsulterReservations (ConsultationRequest) returns (ConsultationResponse);
  rpc AjouterChambre (ChambreRequest) returns (ChambreResponse);
  rpc SupprimerChambre (ChambreRequest) returns (ChambreResponse);
  rpc GetAllChambres (Empty) returns (stream Chambre);
}

message ReservationRequest {
  int32 chambre_id = 1;
  string date_debut = 2;
  string date_fin = 3;
}

message ReservationResponse {
  bool succes = 1;
  string message = 2;
  int32 reservation_id = 3;
}

message DisponibiliteRequest {
  int32 chambre_id = 1;
  string date_debut = 2;
  string date_fin = 3;
}

message DisponibiliteResponse {
  bool disponible = 1;
  string message = 2;
}

message AnnulationRequest {
  int32 reservation_id = 1;
}

message AnnulationResponse {
  bool succes = 1;
  string message = 2;
}

message ConsultationRequest {
  int32 reservation_id = 1;
}

message Reservation {
  int32 reservation_id = 1;
  int32 chambre_id = 2;
  string date_debut = 3;
  string date_fin = 4;
}

message ConsultationResponse {
  repeated Reservation reservations = 1;
}

message ChambreRequest {
  int32 numero = 1;
}

message ChambreResponse {
  bool succes = 1;
  string message = 2;
}

message Empty {}

message Chambre {
  int32 numero = 1;
}